library/general/src/lib/ui/installation/layout.rb

```diff
@@ -0,0 +1,166 @@
+        def with_steps
+          config = LayoutConfig.new.tap { |c| c.mode = LayoutConfig::Mode::STEPS }
+
+          new(config)
+        end
+
+        # Creates a new layout with without a sidebar but with the title of the dialogs on the left
+        #
+        # @return Layout]
+        def with_title_on_left
+          config = LayoutConfig.new.tap { |c| c.mode = LayoutConfig::Mode::TITLE_ON_LEFT }
+
+          new(config)
+        end
+
+        # Creates a new layout with without a sidebar but with the title of the dialogs on top
+        #
+        # @return [Layout]
+        def with_title_on_top
+          config = LayoutConfig.new.tap { |c| c.mode = LayoutConfig::Mode::TITLE_ON_TOP }
+
+          new(config)
+        end
+
+        # Creates a new layout according to the product features
+        #
+        # @see LayoutConfig.from_product_features
+        #
+        # @return [Layout]
+        def from_product_features
+          config = LayoutConfig.from_product_features
+
+          new(config)
+        end
+      end
+
+      # Configures the layout to show a banner
+      def show_banner
+        config.banner = true
+      end
+
+      # Configures the layout to not show a banner
+      def hide_banner
+        config.banner = false
+      end
+
+      # Whether the layout includes a banner
+      #
+      # @return [Boolean]
+      def banner?
+        config.banner
+      end
+
+      # Whether the layout has a left sidebar to place the installation steps
+      #
+      # @return [Boolean]
+      def with_steps?
+        config.mode == LayoutConfig::Mode::STEPS
+      end
+
+      # Whether the layout has the title of the dialogs on the left
+      #
+      # @return [Boolean]
+      def with_title_on_left?
+        config.mode == LayoutConfig::Mode::TITLE_ON_LEFT
+      end
+
+      # Whether the layout has the title of the dialogs on the top
+      #
+      # @return [Boolean]
+      def with_title_on_top?
+        config.mode == LayoutConfig::Mode::TITLE_ON_TOP
+      end
+
+      # Opens a new wizard according to the layout configuration
+      #
+      # @yield Code to run after opening the wizard. The wizard is automatically closed.
+      def open_wizard(&block)
+        Yast::UI.SetProductLogo(banner?)
+
+        case config.mode
+        when LayoutConfig::Mode::STEPS
+          Yast::Wizard.OpenNextBackStepsDialog
+        when LayoutConfig::Mode::TITLE_ON_LEFT
+          Yast::Wizard.OpenLeftTitleNextBackDialog
+        else
+          Yast::Wizard.OpenNextBackDialog
+        end
+
+        return unless block_given?
+
+        block.call
+        close_wizard
+      end
+
+      # Closes the wizard
+      def close_wizard
+        Yast::Wizard.CloseDialog
+      end
+
+    private
+
+      # @return LayoutConfig
+      attr_reader :config
+
+      # Constructor
+      #
+      # @param [Layout, nil] Layout configuration
+      def initialize(config = nil)
+        @config = config || LayoutConfig.new
```

Comment:
well, I would not use nil at all here. Simply I would do  ( and it is also better rendered in yardoc )
```
      def initialize(config = LayoutConfig.new)
          @config = config
```